packages/durable-objects/src/accounts-do.ts

```diff
@@ -0,0 +1,88 @@
+import { DurableObjectState } from '@cloudflare/workers-types';
+import { APIEnv, SentinalRequest } from './types';
+import { unique } from '@notional-finance/helpers';
+
+export class AccountsDO {
+  state: DurableObjectState;
+  env: APIEnv;
+
+  constructor(state: DurableObjectState, env: APIEnv) {
+    this.state = state;
+    this.env = env;
+  }
+
+  async fetch(request: Request): Promise<Response> {
+    try {
+      const req = request.clone();
+      const { method } = request;
+      if (method !== 'POST' && method !== 'GET') {
+        return new Response('Method Not Allowed', { status: 405 });
+      }
+      if (method === 'POST') {
```

Comment:
You will probably need to authenticate the request here, right?

Reply:
I could create some kind of token that we could put in the url as a param possibly

Reply:
We might also be able to check the `Host` header in the request in addition to the token as a second layer of protection. It seems like Host can be faked if Cloudflare chooses not to verify it, but I think you can test if Cloudflare does indeed verify it by trying to curl the endpoint from your local machine with a faked Host header.

Reply:
I looked at the web app firewall briefly, thought maybe we could allow posts only from the sentinel